models/organization/org.go

```diff
@@ -403,7 +403,10 @@ func GetUsersWhoCanCreateOrgRepo(ctx context.Context, orgID int64) ([]*user_mode
 		Join("INNER", "`team_user`", "`team_user`.uid=`user`.id").
 		Join("INNER", "`team`", "`team`.id=`team_user`.team_id").
 		Where(builder.Eq{"team.can_create_org_repo": true}.Or(builder.Eq{"team.authorize": perm.AccessModeOwner})).
-		And("team_user.org_id = ?", orgID).Asc("`user`.name").Find(&users)
+		And("team_user.org_id = ?", orgID).
+		Asc("`user`.name").
+		Distinct("`user`.id").
```

Comment:
Hmm, wasn't there a DB that dislikes `id`? Sqlite or something like that?
I think we should convert it to
```suggestion
		Distinct("`user`.`id`").
```

Reply:
I think you meant PostgreSQL, with the `pq: syntax error at or near "."` error? No clue when you have to escape the column, let's wait for https://drone.gitea.io/go-gitea/gitea/65986/3/8 to find out.

Reply:
And I totally forgot, the integration tests doesn't check for duplicated user IDs, that's the `unit-test` which tests via SQLite... FWIW, I see at least [one other reference](https://github.com/go-gitea/gitea/blob/a1fcb1cfb84fd6b36c8fe9fd56588119fa4377bc/models/organization/org.go#L758) that doesn't escape `id`

Reply:
> Hmm, wasn't there a DB that dislikes `id`? Sqlite or something like that?
> 
> I think we should convert it to
> 
> ```suggestion
> 
> 		Distinct("`user`.`id`").
> 
> ```

I also think not all database support the behaviour. I think we can use a map[int64]*User instead of []*User to resolve the problem.

Reply:
Which behavior? This is nothing new in the Gitea codebase from what I see, `Distinct` is used in the codebase even in combination with `Join`... Am I missing something?

Reply:
Not all databases support select non distinct columns, you can find only distinct columns will be selected in all places.

Reply:
Alright, I wasn't aware that such databases existed 77d56f98ea2f3e70456a362f8feb8f63addf9ea7 addresses it.